src/com/oltpbenchmark/schema/TableSchema.java

```diff
@@ -0,0 +1,65 @@
+package com.oltpbenchmark.schema;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class TableSchema {
+    private final String name;
+    private final List<Column> columns;
+    private final String primaryKey;
+    private final String partitionKey;
+    
+    public TableSchema(String name, List<Column> columns, String primaryKey, String partitionKey) {
+        this.name = name;
+        this.columns = columns;
+        this.primaryKey = primaryKey;
+        this.partitionKey = partitionKey;
+    }
+    
+    public String name() {
+        return name;
+    }
+
+    public List<Column> getColumns() {
+        return columns;
+    }
+
+    public String getPrimaryKey() {
+        return primaryKey;
+    }
+
+    public String getPartitionKey() {
+        return partitionKey;
+    }
+}
+
+class TableSchemaBuilder {
+    private String name = null;
+    private List<Column> columns = new ArrayList<Column>();
+    private String primaryKey = null;
+    private String partitionKey = null;
+    
+    TableSchemaBuilder name(String name) {
+        this.name = name;
+        return this;
+    }
+    TableSchemaBuilder column(String name, String decl) {
+        this.columns.add(new Column(name, decl));
+        return this;
+    }
+    TableSchemaBuilder primaryKey(String primaryKeyDecl) {
+        this.primaryKey = primaryKeyDecl;
+        return this;
+    }
+    TableSchemaBuilder partitionKey(String partitionKey) {
+        this.partitionKey = partitionKey;
+        return this;
+    }
+
+    public TableSchema build() {
+        if (name == null || columns.size() == 0) {
```

Comment:
No need to check `name` here as it is always set via constructor.
Also check columns emptiness with `isEmpty()`

Reply:
Changed to isEmpty() but kept the null check for the name because someone may pass null to the constructor.

Reply:
By following this logic all arguments should be checked for null in java.
In this particular case this name is initialized explicitly, from my point of view null check is not necessary.
In case you are thinking this check should stay you have to make it in constructor.
Also quite common practice in java is to use @NotNull annotation in such cases.

```
class TableSchemaBuilder {
  private final String name;

  public TableSchemaBuilder(@NotNull String name) { 
    this.name = name;
  }
}
```
BTW in case name is checked for null why it is not checked for emptiness?

Reply:
I actually did not know about the NotNull annotation. Added it now, thanks Dmitry. And yes I did add an isEmpty check, but that is in the patch that was submitted after this one, that's why it is not visible here.